### PR TITLE
fix: use semantic elements

### PR DESCRIPTION
**Pull request by mfranzke**
especially for accessibility reasons; to e.g. have those announced correctly by screenreaders and even also make them reachable on keyboard usage.

Resolves https://github.com/code-hike/codehike/issues/335

### CODE REVIEW

**Review by nmerget — APPROVED** [review]

**Comment by pomber** [comment]
thanks, I'll take a look later. 
(maybe you can search for examples of others resetting the button's browser default styles to see if anything is missing)

File: packages/mdx/src/mini-editor/editor-frame.tsx [hunk 1]
```diff
@@ -167,7 +167,7 @@ function TabsContainer({
       )}
       {showFrameButtons ? <FrameButtons /> : <div />}
       {tabs.map(({ title, active, style }) => (
-        <div
+        <button
```
Comment (melanieseltzer) [inline packages/mdx/src/mini-editor/editor-frame.tsx:170]: Just a passing suggestion (maybe not something to implement here in this PR, but just a thought).

I would maybe suggest using an accessible UI foundation package for the editor (which is essentially a Tabs component, I think?). There are quite a few [keyboard interactions](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/) (e.g. roving focus) that are a bit tricky to get right when rolling your own.

For comparison's sake, here are a few that I know of, off the top of my head:

- https://react-spectrum.adobe.com/react-aria/useTabList.html
- https://reach.tech/tabs
- https://reakit.io/docs/tab/
- https://www.radix-ui.com/docs/primitives/components/tabs

**Comment by Mantish** [comment]
Awesome, this is exactly what we need to fix #335

**Comment by pomber** [comment]
Sorry for the delay. I'll try to merge it this week.

File: packages/mdx/src/mini-editor/index.scss [hunk 2]
```diff
@@ -102,4 +104,17 @@
...
+    border: 1px solid transparent;
+    color: inherit;
+  }
+
+  svg {
+    width: 1.5em;
+    height: 1.5em;
+    min-width: 1.5em;
+    min-height: 1.5em;
+  }
```
Comment (Mantish) [inline packages/mdx/src/mini-editor/index.scss:119]: I can see that the copy button is also used with other classes like `.ch-code-button`.
While you can copy these reset styles into the appropriate scss file, another option would be to add the styles directly yo the button on the `copy-buton.tsx` file. Something like this:

```
<button
  style={{
    background: "none",
    border: 0,
    color: "inherit",
    fontSize: "inherit",
    padding: 0,
    ...style,
  }}
  ...
```

File: packages/mdx/src/mini-editor/editor-frame.tsx (hunk 1 above)
Comment (mfranzke) [reply packages/mdx/src/mini-editor/editor-frame.tsx:170]: valid aspect; I've opened a [new issue](https://github.com/code-hike/codehike/issues/344) for this, as I'm not necessarily a React expert and couldn't support on this end.

**Comment by pomber** [comment]
I added the `button` element to a few more buttons and created a `reset-button` mixin. 
Also rolled back the changes to tabs, that's harder and we have [another issue](https://github.com/code-hike/codehike/issues/344) for that.

Thank you all.

**Pull request merged**